File: gerenciador de tarefas/models/Usuario.php
<?php 
class Usuario {
    private $nome;
    private $email;
    private $senha;
    private $listas;

    public function __construct($email, $nome, $senha){
        $this->email = $email;
        $this->nome = $nome;
        $this->senha = $senha;
        $this->listas = array();
    }

    public function getNome() {
        return $this->nome;
    }

    public function getSenha() {
        return $this->senha;
    }

    public function getListas() {
        return $this->listas;
    }

    public function adicionarLista($lista) {
        $this->listas[] = $lista;
    }

    private function getIndiceLista($titulo) {
        foreach ($this->listas as $indice => $lista) {
            if ($lista->getTitulo() === $titulo) {
                return $indice;
            }
        }
        return false; // Retorna false se a lista não for encontrada
    }

    public function excluirLista($lista) {
        // Encontra o índice da lista no array de listas do usuário
        $indice = $this->getIndiceLista($lista->getTitulo());

        // Se o índice for encontrado, remove a lista
        if ($indice !== false) {
            array_splice($this->listas, $indice, 1);
        }
    }
}
?>

File: gerenciador-de-tarefas-spring/gerenciador-de-tarefas/src/main/java/com/tanyrm/gerenciadordetarefas/model/Usuario.java
package model;

import java.util.ArrayList;

public class Usuario {
    private String nome;
    private String email;
    private String senha;
    private ArrayList<Lista> listas;

    public Usuario(String email, String nome, String senha) {
        this.email = email;
        this.nome = nome;
        this.senha = senha;
        this.listas = new ArrayList<Lista>();
    }

    public String getNome() {
        return nome;
    }

    public String getSenha() {
        return senha;
    }

    public ArrayList<Lista> getListas() {
        return listas;
    }

    public void setListas(ArrayList<Lista> listas) {
        this.listas = listas;
    }

    public void adicionarLista(Lista lista) {
        this.listas.add(lista);
    }

    public int getIndiceLista(String titulo) {
        for (int indice = 0; indice < listas.size(); indice++) {
            Lista lista = listas.get(indice);
            if (lista.getTitulo().equals(titulo)) {
                return indice;
            }
        }
        return -1; // Retorna -1 se a lista não for encontrada
    }

    public void excluirLista(Lista lista) {
        int indice = this.getIndiceLista(lista.getTitulo());
        if (indice != -1) {
            listas.subList(indice, indice + 1).clear();
        }
    }
}
